tools/chatgpt/chatgpt.xml

```diff
@@ -0,0 +1,109 @@
+    <configfiles>
+        <configfile name="openai_api_key"><![CDATA[
+$__user__.extra_preferences.get('chatgpt|api_key', "")
+        ]]></configfile>
+    </configfiles>
+    <inputs>
+        <param name="context" type="data" multiple="true" optional="false" format="doc,docx,html,json,pdf,txt,jpg,jpeg,png,webp,gif" label="Context"/>
+        <param name="question" type="text" optional="false" label="Question" help="Question about the text provided"/>
+        <param name="model" type="select" optional="false" label="Model" help="Select the model you want to use">
+            <option value="gpt-4o-mini">Affordable and intelligent small model for fast, lightweight tasks (gpt-4o-mini)</option>
+            <option value="gpt-4o">High-intelligence flagship model for complex, multi-step tasks (gpt-4o)</option>
+            <option value="gpt-4-turbo">The previous set of high-intelligence model with vision capabilities (gpt-4-turbo)</option>
+            <option value="gpt-4">The previous set of high-intelligence model (gpt-4) (not supporting images)</option>
+            <option value="gpt-3.5-turbo">A fast, inexpensive model for simple tasks (GPT-3.5-turbo) (not supporting images)</option>
+        </param>
+    </inputs>
+    <outputs>
+        <data name="output" format="txt" label="${tool.name} on ${on_string}" from_work_dir="./output.txt"/>
+    </outputs>
+    <tests>
+        <test expect_failure="true">
+            <param name="context" value="chatgpt_test.txt" ftype="txt"/>
+            <param name="question" value="What is this?"/>
+            <param name="model" value="gpt-4o-mini"/>
+        </test>
+    </tests>
+    <help><![CDATA[
+
+.. class:: infomark
+
+**What it does**
+
+This tool leverages OpenAI's ChatGPT API to generate responses based on user-provided context and questions.
+Users can upload context data in various formats and ask questions related to that data.
+The tool then processes the data using the selected ChatGPT model, returning an AI-generated response tailored to the context provided.
+
+To utilize this tool, users need to input their OpenAI API key in the user preferences. To obtain an API key, visit https://platform.openai.com/account/api-keys.
+
+When you upload a file and ask a question, the tool sends the file to OpenAI's servers using their API. 
+OpenAI's models process the data and generate a response based on the context and question provided. 
+After receiving the response, the tool returns it to the user. 
+The files you upload are then securely deleted from the OpenAI's server, 
+ensuring that your data is handled responsibly and is not stored beyond its necessary use. 
+If the tool fails to delete your uploaded files automatically, you can manually delete them by visiting https://platform.openai.com/storage/.
+
+Usage
+.....
+
+**Input**
+
+1. **Upload Context Data**: Users can upload up to 500 files in formats such as DOC, DOCX, HTML, JSON, PDF, TXT, JPG, JPEG, PNG, WEBP, or GIF. 
```

Comment:
If that is true, set a max="500" to the input. You can try if this works with something small like max="4"